Products/CMFPlone/tests/test_recyclebin.py

```diff
@@ -0,0 +1,387 @@
+from datetime import datetime
+from datetime import timedelta
+from persistent.mapping import PersistentMapping
+from Products.CMFPlone.recyclebin import ANNOTATION_KEY
+from Products.CMFPlone.recyclebin import RecycleBin
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import unittest
+
+
+class TestRecycleBin(unittest.TestCase):
+    """Test the RecycleBin functionality"""
```

Comment:
I would like the test suite to not depend on mocks so heavily. That tends to end up testing a made-up situation rather than the real behavior of the complete system. I want functional tests that do things with an actual Plone site (using the existing integration test or functional test fixtures).

Reply:
done in https://github.com/plone/Products.CMFPlone/pull/4168/commits/ec0e0f25910fd16812b49336d624149b4943a083